metricbeat/module/logstash/node_stats/data_xpack.go

```diff
@@ -213,26 +213,21 @@ func makeClusterToPipelinesMap(pipelines []PipelineStats, overrideClusterUUID st
 	var clusterToPipelinesMap map[string][]PipelineStats
 	clusterToPipelinesMap = make(map[string][]PipelineStats)
 
-	if overrideClusterUUID != "" {
-		clusterToPipelinesMap[overrideClusterUUID] = pipelines
-		return clusterToPipelinesMap
-	}
-
 	for _, pipeline := range pipelines {
-		var clusterUUIDs []string
+		clusterUUIDs := make(map[string]struct{}, 0)
```

Comment:
this is `common.StringSet`.

The overall data expension starts by converting the `map[string]pipelineState` into a `[]pipelineState`. Would it make sense to keep the map instead, to reduce the chance of creating duplicate entries in general?

Reply:
I've changed the code to use `common.StringSet` in 4cb337f96 but I didn't follow this:

> The overall data expension starts by converting the `map[string]pipelineState` into a `[]pipelineState`. Would it make sense to keep the map instead, to reduce the chance of creating duplicate entries in general?

I think it's because I'm not finding the `map[string]pipelineState` you mentioned. Where are you seeing that?

Reply:
We do the [conversion here](https://github.com/ycombinator/beats/blob/mon-ls-dup-cluster-uuid/metricbeat/module/logstash/node_stats/data_xpack.go#L174).
I was wondering if it is necessary, or maybe can even lead to similar problems (in the future) if we are not careful about not producing duplicates.